handler.go

```diff
@@ -1,99 +1,148 @@
 package main
 
 import (
+	"errors"
 	"fmt"
+	"io/ioutil"
 	"log"
 	"net/http"
 	"strings"
 )
 
-func handler(w http.ResponseWriter, r *http.Request) {
-	if r.URL.Path != "/" && !strings.HasSuffix(r.URL.Path, ".svg") {
-		http.NotFound(w, r)
-		return
-	}
-
-	logf("----------------------")
-	logf(" => handling request:  %v", r.URL)
-	logf("----------------------")
+func analysisSetup() (r renderOpts) {
+	r = renderOpts{
+		focus:   *focusFlag,
+		group:   []string{*groupFlag},
+		ignore:  []string{*ignoreFlag},
+		include: []string{*includeFlag},
+		limit:   []string{*limitFlag},
+		nointer: *nointerFlag,
+		nostd:   *nostdFlag}
 
-	focus := *focusFlag
-	nostd := *nostdFlag
-	nointer := *nointerFlag
-	group := *groupFlag
-	limit := *limitFlag
-	ignore := *ignoreFlag
-	include := *includeFlag
-
-	if f := r.FormValue("f"); f == "all" {
-		focus = ""
-	} else if f != "" {
-		focus = f
-	}
-	if std := r.FormValue("std"); std != "" {
-		nostd = false
-	}
-	if inter := r.FormValue("nointer"); inter != "" {
-		nointer = true
-	}
-	if g := r.FormValue("group"); g != "" {
-		group = g
-	}
-	if l := r.FormValue("limit"); l != "" {
-		limit = l
-	}
-	if ign := r.FormValue("ignore"); ign != "" {
-		ignore = ign
-	}
-	if inc := r.FormValue("include"); inc != "" {
-		include = inc
-	}
+	return r
+}
 
+func processListArgs(r *renderOpts) (e error) {
 	var groupBy []string
-	for _, g := range strings.Split(group, ",") {
+	for _, g := range strings.Split(r.group[0], ",") {
 		g := strings.TrimSpace(g)
 		if g == "" {
 			continue
 		}
 		if g != "pkg" && g != "type" {
-			http.Error(w, "invalid group option", http.StatusInternalServerError)
+			e = errors.New("invalid group option")
 			return
 		}
 		groupBy = append(groupBy, g)
 	}
+	r.group = groupBy
 
 	var ignorePaths []string
-	for _, p := range strings.Split(ignore, ",") {
+	for _, p := range strings.Split(r.ignore[0], ",") {
 		p = strings.TrimSpace(p)
 		if p != "" {
 			ignorePaths = append(ignorePaths, p)
 		}
 	}
+	r.ignore = ignorePaths
 
 	var includePaths []string
-	for _, p := range strings.Split(include, ",") {
+	for _, p := range strings.Split(r.include[0], ",") {
 		p = strings.TrimSpace(p)
 		if p != "" {
 			includePaths = append(includePaths, p)
 		}
 	}
+	r.include = includePaths
 
 	var limitPaths []string
-	for _, p := range strings.Split(limit, ",") {
+	for _, p := range strings.Split(r.limit[0], ",") {
 		p = strings.TrimSpace(p)
 		if p != "" {
 			limitPaths = append(limitPaths, p)
 		}
 	}
+	r.limit = limitPaths
+
+	return
+}
 
-	opts := renderOpts{
-		focus:   focus,
-		group:   groupBy,
-		ignore:  ignorePaths,
-		include: includePaths,
-		limit:   limitPaths,
-		nointer: nointer,
-		nostd:   nostd,
+func outputDot(fname string, outputSvg, outputPng bool) {
```

Comment:
The `handler.go` file is intended to be used for HTTP handlers, but this is not used in HTTP server at all.